epinorm/norm.py

```diff
@@ -282,24 +386,65 @@ def _get_location_from_strain(self, record):
                 return token
 
     def _format_location(self, record):
+        """
+        basically we combine the information from original_record_location_description and the extracted information
+        of the strain. We also clean them (removing unwanted characters)
+
+        we return a dictionaly in json format with all the fields
+        {
+            country
+            areas
+        }
+        """
+
         location = json.loads(record["original_record_location_description"])
-        extracted_location = record["extracted_location"]
-        if "area" not in location:
-            location["area"] = extracted_location
-        country = str(coalesce(location["country"], ""))
-        area = str(coalesce(location["area"], ""))
-        places = []
-        for place in area.split(","):
+
+
+        # format country to make comparison easier
+        location["country"] = location["country"].strip().lower()
+
+        newPlaces = [] # combine areas and extrated location together in this list
+
+        # format areas
+        for place in location["areas"]:
             place = place.replace("_", " ")
             place = re.sub(r"\(.+\)", "", place)
             place = re.sub(r"[^\w\s-]+", "", place)
             place = place.strip()
-            if not place or place == country:
-                continue
-            places.append(place)
-        if places:
-            return ", ".join(places) + ", " + country
-        return country
+            place = place.lower() # for easier comparison later
+
+            if place != "" and place != location["country"] and place not in newPlaces:
+                newPlaces.append(place)
+
+
+        # format extracted field
+        if record["extracted_location"]:
+            place = record["extracted_location"].replace("_", " ")
+            place = re.sub(r"\(.+\)", "", place)
+            place = re.sub(r"[^\w\s-]+", "", place)
+            place = place.strip()
+            place = place.lower() # for easier comparison later
```

Comment:
Some of these lines can be combined I think.
Maybe also the 2 regular expressions.

```suggestion
            place = record["extracted_location"].replace("_", " ").strip().lower()
            place = re.sub(r"\(.+\)", "", place)
            place = re.sub(r"[^\w\s-]+", "", place)
```